lib/git/failed_error.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'git/git_execute_error'
+
+module Git
+  # This error is raised when a git command fails
+  #
+  # The git command executed, status, stdout, and stderr are available from this
+  # object. The #message includes the git command, the status of the process, and
+  # the stderr of the process.
+  #
+  # @api public
+  #
+  class FailedError < Git::GitExecuteError
+    # Create a FailedError object
+    #
+    # @example
+    #   `exit 1` # set $? appropriately for this example
+    #   result = Git::CommandLineResult.new(%w[git status], $?, '', "failed")
+    #   error = Git::FailedError.new(result)
+    #   error.message #=>
+    #     "[\"git\", \"status\"]\nstatus: pid 89784 exit 1\nstderr: \"failed\""
+    #
+    # @param result [Git::CommandLineResult] the result of the git command including
+    #   the git command, status, stdout, and stderr
+    #
+    def initialize(result)
+      super("#{result.git_cmd}\nstatus: #{result.status}\nstderr: #{result.stderr.inspect}")
```

Comment:
@jcouball I think stdout also has to be printed here since stderr is redirected to stdout?? https://github.com/ruby-git/ruby-git/blob/5b0e1c8b62188e2a80427bba3f625aaaf84e7350/lib/git/lib.rb#L1194